torch/autograd/_functions/tensor.py

```diff
@@ -18,9 +18,8 @@ def forward(self, i):
         return result
 
     def backward(self, grad_output):
-        # TODO: this won't have to be zeroed
         grad_input = grad_output.new(self.input_size).zero_()
-        grad_input.index(self.index).copy_(grad_output)
+        grad_input._set_index(self.index, grad_output)
```

Comment:
LongTensor-in-a-tuple case will now fail in forward, because of "Fall back to indexing compatible with numpy". This will work for sure - if there's an index that can't copy the grad_output into grad_input, then `_set_index` will fail. Both `.index` and `._set_index` use the same logic in C.